### PR TITLE
build: add strategies to protostar

### PR DESCRIPTION
**Pull request by tdelabro**
feat: move init loop in composable stategy

### CODE REVIEW

File: contracts/onlydust/marketplace/core/assignment_strategies/composite.cairo [hunk 1]
```diff
@@ -116,4 +113,30 @@ namespace internal {
...
+            contract,
+            class_hs=strategy_hash,
+            calldata_len=strategy_calldata_len,
+            calldata=calldata + 2,
+        );
+        assignment_strategy__composite__strategy_hash_by_index.write(count, strategy_hash);
+
+        let calldata_len = calldata_len - (strategy_calldata_len + 2);
+        let calldata = calldata + strategy_calldata_len + 2;
+        return internal.rec_init_from_call_data(contract, count + 1, calldata_len, calldata);
```
[the remark this answers is not in the document]
Comment (tdelabro) [reply contracts/onlydust/marketplace/core/assignment_strategies/composite.cairo]: done

File: contracts/onlydust/marketplace/core/assignment_strategies/composite.cairo [hunk 2]
```diff
@@ -116,4 +113,30 @@ namespace internal {
...
+    ) -> felt {
+        if (calldata_len == 0) {
+            return (count);
+        }
+
+        let strategy_hash = calldata[0];
+        let strategy_calldata_len = calldata[1];
+
+        // TODO: check strat is allowd
+        // IOnlyDust.assert_hash_allowed(only_dust_contract, class_hash);
```
[the remark this answers is not in the document]
Comment (tdelabro) [reply contracts/onlydust/marketplace/core/assignment_strategies/composite.cairo]: also done

File: scripts/deploy.sh [hunk 3]
```diff
@@ -6,7 +6,7 @@ ROOT=`readlink -f $SCRIPT_DIR/..`
 CACHE_FILE_BASE=$ROOT/scripts/configuration/deployed_contracts
 STARKNET_ACCOUNTS_FILE=$HOME/.starknet_accounts/starknet_open_zeppelin_accounts.json
 PROTOSTAR_TOML_FILE=$ROOT/protostar.toml
-STARKNET_VERSION="0.10.0"
+STARKNET_VERSION="0.10.1"
```
Comment (AnthonyBuisset) [inline scripts/deploy.sh:9]: IIRC this version is broken

File: contracts/onlydust/marketplace/core/assignment_strategies/test_composite.cairo [hunk 4]
```diff
@@ -36,20 +36,14 @@ func __setup__{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
...
+    let composite_strategy_hash = Composite.default();
     with composite_strategy_hash {
-        Composite.initialize(3, strategies);
-
         let (strategies_len, strategies) = Composite.strategies();
         assert 3 = strategies_len;
-        assert 0x1111 = strategies[0];
-        assert 0x2222 = strategies[1];
-        assert 0x3333 = strategies[2];
+        assert test_strategy_hash = strategies[0];
```
Comment (AnthonyBuisset) [inline contracts/onlydust/marketplace/core/assignment_strategies/test_composite.cairo:44]: the goal of this test was to verify we could add 3 different strategies and store them/retrieve them in the same order

**Review by AnthonyBuisset — APPROVED** [review]

**Pull request merged**